### PR TITLE
[opt](inverted index) prevent excessive memory usage caused by too many terms.

### PR DESCRIPTION
**Pull request by zzzxl1993**
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/olap/rowset/segment_v2/inverted_index/query/regexp_query.cpp [hunk 1]
```diff
@@ -30,80 +30,43 @@ RegexpQuery::RegexpQuery(const std::shared_ptr<lucene::search::IndexSearcher>& s
...
-        return;
-    }
-
-    auto on_match = [](unsigned int id, unsigned long long from, unsigned long long to,
-                       unsigned int flags, void* context) -> int {
-        *((bool*)context) = true;
-        return 0;
-    };
+    re2::RE2 pattern(terms[0]);
+    std::vector<std::string> results;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index/query/regexp_query.cpp:39]: The original name terms is more easy to understand.

File: be/src/olap/rowset/segment_v2/inverted_index/query/disjunction_query.h [hunk 2]
```diff
@@ -28,17 +28,20 @@ class DisjunctionQuery : public Query {
...
     void add(const std::wstring& field_name, const std::vector<std::string>& terms) override;
     void search(roaring::Roaring& roaring) override;
 
 private:
     std::shared_ptr<lucene::search::IndexSearcher> _searcher;
 
-    std::vector<Term*> _terms;
-    std::vector<TermDocs*> _term_docs;
-    std::vector<TermIterator> _term_iterators;
+    // std::vector<Term*> _terms;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index/query/disjunction_query.h:39]: just delete

File: be/src/olap/rowset/segment_v2/inverted_index/query/disjunction_query.h [hunk 3]
```diff
@@ -28,17 +28,20 @@ class DisjunctionQuery : public Query {
...
+    // std::vector<TermDocs*> _term_docs;
+    // std::vector<TermIterator> _term_iterators;
+
+    std::wstring _field_name;
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index/query/disjunction_query.h:43]: It's better to put common fields: _filed_name and _terms in parent class Query.

File: be/src/olap/rowset/segment_v2/inverted_index/query/disjunction_query.cpp [hunk 4]
```diff
@@ -23,58 +23,46 @@ DisjunctionQuery::DisjunctionQuery(const std::shared_ptr<lucene::search::IndexSe
...
         } else {
             roaring |= result;
         }
     };
-    for (int i = 0; i < _term_iterators.size(); i++) {
-        auto& iter = _term_iterators[i];
+    for (int i = 0; i < _terms.size(); i++) {
         if (i == 0) {
-            func(iter, true);
+            func(_terms[i], true);
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/inverted_index/query/disjunction_query.cpp:63]: use one call func(_terms[i], i == 0) instead of if else

**Review by xiaokang — COMMENTED** [review]
pls add testcase

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**